store/store.go

```diff
@@ -415,39 +419,37 @@ func (s *store) IsPruned() bool {
 	return s.isPruned
 }
 
-func (s *store) Prune(resultFunc func(pruned, skipped, pruningHeight uint32)) error {
+func (s *store) Prune(resultFunc func(pruned bool, pruningHeight uint32) bool) error {
+	s.lk.Lock()
+	defer s.lk.Unlock()
+
 	cert := s.lastCertificate()
 
-	// at genesis block
+	// Store is at the genesis height
 	if cert == nil {
 		return nil
 	}
 
 	retentionBlocks := s.config.RetentionBlocks()
-
 	if cert.Height() < retentionBlocks {
 		return nil
 	}
 
 	pruningHeight := cert.Height() - retentionBlocks
-
 	for i := pruningHeight; i >= 1; i-- {
 		deleted, err := s.pruneBlock(i)
 		if err != nil {
 			return err
 		}
 
-		if err := s.WriteBatch(); err != nil {
+		if err := s.writeBatch(); err != nil {
 			return err
 		}
 
-		if deleted {
-			resultFunc(1, 0, i)
-
-			continue
+		if resultFunc(deleted, i) {
```

Comment:
Does not required return bool and check result func.

I think remove return bool.

Reply:
`resultFunc` is not a very good name here; it checks if the loop should continue or not. This prevents a random crash if the user presses Ctrl+C while Prune is working.